src/robusta/core/sinks/robusta/dal/supabase_dal.py

```diff
@@ -537,3 +541,43 @@ def publish_cluster_nodes(self, node_count: int):
             self.handle_supabase_error()
 
         logging.info(f"cluster nodes: {UPDATE_CLUSTER_NODE_COUNT} => {data}")
+
+    def get_account_resources(self) -> List[AccountResource]:
+        res = (
+            self.client.table(ACCOUNT_RESOURCE_TABLE)
+            .select("entity_id", "resource_kind", "clusters_target_set", "resource_state", "deleted", "updated_at")
+            .filter("account_id", "eq", self.account_id)
+            .filter("deleted", "eq", False)
+            .execute()
+        )
+        if res.get("status_code") not in [200]:
+            msg = f"Failed to get existing account resources (supabase) error: {res.get('data')}"
+            logging.error(msg)
+            self.handle_supabase_error()
+            raise Exception(msg)
+
+        account_resources: List[AccountResource] = []
+        for data in res.get("data"):
+            if not data["resource_state"]:
+                continue
+
+            resource_state: Optional[ResourceState] = None
+            if data["resource_kind"] == ResourceKind.PrometheusAlert:
+                if "*" in data["clusters_target_set"] or self.cluster in data["clusters_target_set"]:
```

Comment:
The `clusters_target_set` be a part of the query. You don't have to fetch records for all clusters